onnx/checker.h

```diff
@@ -9,6 +9,19 @@ namespace checker {
 class ValidationError : public std::runtime_error {
  public:
   using std::runtime_error::runtime_error;
+  const char* what() const noexcept override {
+    if (!expanded_message_.empty()) {
+      return expanded_message_.c_str();
+    }
+    return std::runtime_error::what();
+  }
+  void AppendContext(const std::string& context) {
+    expanded_message_ = onnx::MakeString(
+        std::runtime_error::what(), "\n\n==> Context: ", context);
+  }
+
+ private:
+  std::string expanded_message_;
```

Comment:
How about making it a vector so each level can put its own helpful debug message?

Reply:
Each level can call AppendContext - they will be appended continuously